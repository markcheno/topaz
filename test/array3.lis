#    0: adjs		2
#    5: pushstring	'Mark Chenoweth'
#   25: setglobal	0
#   30: pushint		0
#   35: setglobal	1
#   40: getglobal	1
#   45: pushint		0
#   50: getglobal	0
#   55: pushint		1
#   60: call		len
#   65: leq
#   66: jnz		97
#   71: jmp		152
#   76: getglobal	1
#   81: pushint		1
#   86: add
#   87: setglobal	1
#   92: jmp		40
#   97: pushint		0
#  102: pushstring	'name[%d]=%c
'
#  120: getglobal	1
#  125: getglobal	0
#  130: getglobal	1
#  135: getarray
#  136: pushint		3
#  141: call		printf
#  146: pop
#  147: jmp		76
#  152: pushint		0
#  157: getglobal	0
#  162: pushint		20
#  167: getarray
#  168: pushint		1
#  173: call		println
#  178: pop
#  179: halt
