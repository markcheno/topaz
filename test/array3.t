
name = "Mark Chenoweth"

for( x=0; x<=len(name); x=x+1 )
  printf("name[%d]=%c\n",x,name[x])
end

println(name[20])
