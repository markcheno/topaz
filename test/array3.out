name[0]=M
name[1]=a
name[2]=r
name[3]=k
name[4]= 
name[5]=C
name[6]=h
name[7]=e
name[8]=n
name[9]=o
name[10]=w
name[11]=e
name[12]=t
name[13]=h
runtime error - index out of bounds: 14
